src/allocator.rs

```diff
@@ -635,15 +659,15 @@ impl Allocator {
         }
         if counter != new_size {
             self.u8_vec.truncate(start);
-            return err(
+            return Err(EvalErr::InternalError(
                 self.nil(),
-                "(internal error) concat passed invalid new_size",
-            );
+                "concat passed invalid new_size".to_string(),
```

Comment:
If we stick with `String`, we could include `new_size` and `counter` in the message.